copier/cli.py

```diff
@@ -191,6 +193,25 @@ def data_switch(self, values: StrSeq) -> None:
             key, value = arg.split("=", 1)
             self.data[key] = value
 
+    @cli.switch(
+        ["--data-file"],
+        str,
+        help="Load data from a YAML file",
+        excludes=["--data"],
+    )
+    def data_file_switch(self, path: str) -> None:
+        """Update [data][] with provided values.
+
+        Arguments:
+            path: The path to the YAML file to load.
+        """
+        if not Path(path).exists():
+            raise UserMessageError(f"Data file {path} not found")
```

Comment:
You can use some plumbum helpers:

```suggestion
    @cli.switch(
        ["--data-file"],
        cli.ExistingPath,
        help="Load data from a YAML file",
        excludes=["--data"],
    )
    def data_file_switch(self, path: plumbum.LocalPath) -> None:
        """Update [data][] with provided values.

        Arguments:
            path: The path to the YAML file to load.
        """
```